packages/angular/cli/src/command-builder/command-module.ts

```diff
@@ -129,6 +130,14 @@ export abstract class CommandModule<T extends {} = {}> implements CommandModuleI
       await this.reportAnalytics(camelCasedOptions);
     }
 
+    // Set up autocompletion if appropriate.
+    const autocompletionExitCode = await considerSettingUpAutocompletion(this.context.logger);
```

Comment:
Maybe placing this in the command-runner prior to running the command is better. Also, we might want to disable the auto completion prompt for certain commands, example, `ng update`.

Reply:
I saw that for analytics, but didn't fully understand it. Thinking now, is that intended to avoid:

1. User runs `ng update` on 13.3.x.
2. `@angular/cli@13.3.x` updates itself to `14.0.x`.
3. `@angular/cli@14.0.x` runs `ng update`.
4. `ng update` prompts to set up autocompletion.

Instead we would rather prompt on the next `ng build` / `ng test` right? I updated it to ignore `ng update`, just want to make sure I understand the motivation.

I also skipped the prompt for `ng completion` commands, since that would be redundant. Calling `ng completion` means you already have a setup, and `ng completion init` means you _want_ a setup, so a prompt is unnecessary in both cases.

Reply:
Yeah, that is indeed the motivation behind that.